tests/compiler/assert-nonnull.untouched.wat

```diff
@@ -114,6 +114,10 @@
   local.get $1
   call $~lib/array/Array<assert-nonnull/Foo>#__unchecked_get
   local.set $2
+  i32.const 1
+  drop
+  i32.const 1
+  drop
```

Comment:
It looks as some unnecessary noise. Could we avoid this?

Reply:
These are from being super careful when ignoring a branch due to a constant condition, making sure to always execute the condition nonetheless. Now this isn't exactly necessary currently because all we are looking for are constants without side effects, but once we can evaluate stuff with side effects, doing so will avoid any issues.

I am currently somewhat shifting from doing trivial optimizations on untouched output to just emitting correct code in order to avoid the possibility of introducing errors, as special cases turned out to introduce subtle bugs in the past, like missing an autorelease due to shortcutting. One could argue that the code now exactly represents what would execute in a user's code before optimizations, but after precomputing. The latter is only necessary to provide constant branch elimination in the first place, as skipping that would essentially remove the feature.